src/main/java/de/hysky/skyblocker/skyblock/dungeon/GoldorWaypointsManager.java

```diff
@@ -0,0 +1,252 @@
+                case TERMINAL:
+                    TERMINALS.add(waypoint);
+                    break;
+                case DEVICE:
+                    DEVICES.add(waypoint);
+                    break;
+                case LEVER:
+                    LEVERS.add(waypoint);
+                    break;
+            }
+        }));
+    }
+
+    /**
+     * Checks if we should process messages
+     *
+     * @return true if we should process messages
+     */
+    private static boolean shouldProcessMsgs() {
+        return (loaded && SkyblockerConfigManager.get().dungeons.goldor.enableGoldorWaypoints && Utils.isInDungeons() && DungeonManager.isInBoss() && DungeonManager.getBoss().isFloor(7));
+    }
+
+    /**
+     * Given a list of waypoints to operate on and a player name, hide the visible waypoint that is closest to the player
+     *
+     * @param waypoints  The list of waypoints to operate on
+     * @param playerName The name of the player to check against
+     */
+    private static void RemoveNearestWaypoint(ObjectArrayList<GoldorWaypoint> waypoints, String playerName) {
+        MinecraftClient client = MinecraftClient.getInstance();
+        if (client.world == null) return;
+
+        Optional<Vec3d> posOptional = client.world.getPlayers().stream().filter(player -> player.getGameProfile().getName().equals(playerName)).findAny().map(Entity::getPos);
+
+        if (posOptional.isPresent()) {
+            Vec3d pos = posOptional.get();
+
+            waypoints.stream().filter(GoldorWaypoint::shouldRender).min((a, b) -> {
+                double distA = a.centerPos.distanceTo(pos);
+                double distB = b.centerPos.distanceTo(pos);
+                return Double.compare(distA, distB);
+            }).map(waypoint -> {
+                waypoint.setShouldRender(false);
+                return null;
+            });
+        }
+    }
+
+    /**
+     * Resets state, disabling waypoint rendering (but setting all waypoints to be ready for the next run)
+     */
+    private static void reset() {
+        active = false;
+        currentPhase = 0;
+        enableAll(TERMINALS);
+        enableAll(DEVICES);
+        enableAll(LEVERS);
+    }
+
+    /**
+     * Enables rendering for all waypoints in a set
+     *
+     * @param waypoints The set of waypoints to enable rendering for
+     */
+    private static void enableAll(ObjectArrayList<GoldorWaypoint> waypoints) {
+        waypoints.forEach(waypoint -> waypoint.setShouldRender(true));
+    }
+
+    /**
+     * Convenience method to extract the player name from a message
+     *
+     * @param matcher The matcher to extract the name from
+     * @return The player name, or null if the matcher didn't match
+     */
+    @Nullable
+    private static String getPlayerName(Matcher matcher) {
+        return matcher.matches() ? matcher.group("name") : null;
+    }
+
+    private static void onChatMessage(Text text, boolean overlay) {
+        if (!shouldProcessMsgs()) return;
```

Comment:
add a check for `overlay` being false since you'll want to ignore action bar updates